packages/rs-drive-abci/src/validation/state_transition/context.rs

```diff
@@ -0,0 +1,9 @@
+use dpp::dashcore::TxOut;
+
+/// Validation context to share data required for validation between validation functions
+/// It allows to avoid fetching the same data multiple times
+#[derive(Debug, Default)]
+pub struct ValidationDataShareContext {
```

Comment:
We can remove it, otherwise you need to fetch asset lock transaction twice

Reply:
No you don't, you just need to fetch asset lock transactions once when you transform into a state transition action, core signature validation should be there, and only there.

Reply:
But then if we call `validate_identity_and_signatures` we won't validate the ST signature. It doesn't make sense and it will create problems and confusions in future